### PR TITLE
fix: some of the response types

### PR DESCRIPTION
**Pull request by kallberg**
Hello.

Noticed the conversation on the pull-request for version 22 and decided to give it a try. I realized later when I looked at the network graph that you were already in the process of updating using version-22 branch.

I rebased and squashed commits to match your banch.

Don't know what the state of your local branches are so maybe there is a lot of duplicate work and if that is the case please feel free to ignore/close this PR.

Otherwise here's some changes that should fix some of the response type parsing for v22. I don't know if I got all of them.

tried to get most of the types fixed probably missed some

also:
- update dependency openapiv3 (Not sure if there is a reason not to update if so I can revert this)
  - will not output bearerFormat of 'null' and will instead omit bearerFormat as it must be string if included.
- applied cargo clippy fixes
- changed many of the response type parsing strings
- had to introduce new/alternate parsing of set, maps, and list types
- removed parsing of types that are no longer in the html
- adjusted yq to output yaml where newer defaults to json (mikefarah yq v4.35.2)

### CODE REVIEW

File: keycloak-openapi-transformer/src/components/schemas.rs [hunk 1]
```diff
@@ -32,58 +33,71 @@ pub fn parse_schemas(
...
+    if !raw_type.starts_with(MAP_PREFIX) {
+        return None;
     }
+
+    let inner_type_str = raw_type.strip_prefix(MAP_PREFIX)?.strip_suffix(']')?;
+
+    let inner_type = parse_type(inner_type_str);
+
+    let map = openapiv3::Type::Object(ObjectType {
+        additional_properties: Some(openapiv3::AdditionalProperties::Schema(Box::new(
```
Comment (kallberg) [inline keycloak-openapi-transformer/src/components/schemas.rs:100]: Also not sure about this structure. It was hard to tell if this is the intended format. I would need to compare with the actual responses from keycloak api to make sure

**Comment by ccouzens** [comment]
Hey - sorry I saw this too late.

As you may have noticed from me taking months to get to version 22 I don't currently have much time for this. Would you be interested in being a co-maintainer on this project? You'd be doing me a favour.

Chris

**Comment by kallberg** [comment]
Hi Chris,

I'm interested. At least for the next couple of months as the the specs made available thanks to this project will come in handy for me soon. Was considering maintaining a fork.

Erik

**Comment by ccouzens** [comment]
Hey Erik,

You should now have access.

**Comment by kallberg** [comment]
Got it!

I'll send you an email tomorrow with some basic questions about how you'd like us to collaborate

**Pull request merged**